### PR TITLE
Fix/4882 make activities more robust

### PR DESCRIPTION
**Pull request by raphj**
#4882

## Checkpoints

**For developers**

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [X] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [X] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [X] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [X] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Comment by PhilippeAccorsi** [comment]
I have stopped Travis test about concourse-ci (it is not necessary to test in travis).
What can test can be done ? How test this fix ?

**Comment by raphj** [comment]
@PhilippeAccorsi 

- Have two users, sharing one space.
- One user creates a file/notee in the shared space and comments it
- Then this user deletes the file, or moves it a space that the other user cannot access
- see if the activity list of the second user is stuck or not

File: frontend/src/component/Activity/ActivityList.jsx [hunk 1]
```diff
@@ -159,11 +161,11 @@ const ActivityList = (props) => {
...
       )}
       <div className='activityList__list' data-cy='activityList__list'>
-        {(props.activity.list.length > 0
-          ? props.activity.list.filter(activityDisplayFilter).map(renderActivityComponent)
+        {(activities.length > 0
+          ? activities
           : (
             <div className='activityList__placeholder'>
-              {props.activity.hasNextPage ? props.t('Loading recent activities…') : props.t('No activity')}
+              {props.activity.loaded ? props.t('No activity') : props.t('Loading recent activities…')}
```
Comment (raphj) [inline frontend/src/component/Activity/ActivityList.jsx:168]:
```suggestion
              {props.activity.hasNextPage ? props.t('Loading recent activities…') : props.t('No activity')}
```

File: frontend/src/reducer/userActivity.js [hunk 2]
```diff
@@ -20,7 +21,7 @@ const userActivity = (state = defaultUserActivityState, action) => {
     case `${RESET}/${USER_ACTIVITY}`:
       return defaultUserActivityState
     case `${SET}/${USER_ACTIVITY}/${LIST}`:
-      return { ...state, list: action.activityList }
+      return { ...state, list: action.activityList, loaded: true }
```
Comment (raphj) [inline frontend/src/reducer/userActivity.js:24]:
```suggestion
      return { ...state, list: action.activityList }
```

File: frontend/src/reducer/workspaceActivity.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ import { setActivityEventList } from '../util/activity.js'
 
 const defaultWorkspaceActivityState = {
   list: [],
+  loaded: false,
```
Comment (raphj) [inline frontend/src/reducer/workspaceActivity.js:14]:
```suggestion
```

File: frontend/src/reducer/workspaceActivity.js [hunk 4]
```diff
@@ -20,7 +21,7 @@ const workspaceActivity = (state = defaultWorkspaceActivityState, action) => {
     case `${RESET}/${WORKSPACE_ACTIVITY}`:
       return defaultWorkspaceActivityState
     case `${SET}/${WORKSPACE_ACTIVITY}/${LIST}`:
-      return { ...state, list: action.activityList }
+      return { ...state, list: action.activityList, loaded: true }
```
Comment (raphj) [inline frontend/src/reducer/workspaceActivity.js:24]:
```suggestion
      return { ...state, list: action.activityList }
```

File: frontend/src/reducer/userActivity.js [hunk 5]
```diff
@@ -11,6 +11,7 @@ import { setActivityEventList } from '../util/activity.js'
 
 const defaultUserActivityState = {
   list: [],
+  loaded: false,
```
Comment (raphj) [inline frontend/src/reducer/userActivity.js:14]:
```suggestion
```

**Review by gbonaspetti — APPROVED** [review]
ok for frontend part

**Comment by charline-rageade** [comment]
It's ok for review ? Boxes are not checked.
Issue #4882 tested on branch

**Comment by raphj** [comment]
We are waiting for the review of the backend part.

**Comment by charline-rageade** [comment]
I've assigned @grignards & @inkhey for review

File: backend/tracim_backend/tests/functional/test_workspaces.py [hunk 6]
```diff
@@ -5275,10 +5275,7 @@ def test_api__get_content_path__ok__200__nominal_case(
         web_testapp.authorization = ("Basic", ("test@test.test", "test@test.test"))
 
         my_file_path = web_testapp.get(
-            "/api/workspaces/{}/contents/{}/path".format(
-                workspace.workspace_id, my_file.content_id
-            ),
-            status=200,
+            "/api/contents/{}/path".format(my_file.content_id), status=200,
```
Comment (inkhey) [inline backend/tracim_backend/tests/functional/test_workspaces.py:5278]: Can you explain the behavior if content is not accessible to the connected user ?
A test may be nice.

File: frontend/src/component/Activity/ActivityList.jsx [hunk 7]
```diff
@@ -146,6 +146,8 @@ const ActivityList = (props) => {
       )
   }
 
+  const activities = props.activity.list.filter(activityDisplayFilter).map(renderActivityComponent)
```
[the remark this answers is not in the document]
Comment (raphj) [reply frontend/src/component/Activity/ActivityList.jsx:149]: done, thanks

File: backend/tracim_backend/tests/functional/test_workspaces.py (hunk 6 above)
Comment (raphj) [reply backend/tracim_backend/tests/functional/test_workspaces.py:5278]: Good point, I added a test (it returns 400 Content not found in database) @inkhey

**Review by grignards — APPROVED** [review]

File: backend/tracim_backend/tests/functional/test_workspaces.py [hunk 8]
```diff
@@ -5310,6 +5310,51 @@ def test_api__get_content_path__ok__200__nominal_case(
...
+            "test@test.test",
+            password="test@test.test",
+            do_save=True,
+            do_notify=False,
+            profile=profile,
+        )
+        transaction.commit()
+        web_testapp.authorization = ("Basic", ("test@test.test", "test@test.test"))
+
+        web_testapp.get("/api/contents/{}/path".format(my_file.content_id), status=400)
```
Comment (inkhey) [inline backend/tracim_backend/tests/functional/test_workspaces.py:5354]: There is no specific error code returned ? If so, check it.

**Review by inkhey — APPROVED** [review]

**Comment by charline-rageade** [comment]
@raphj I think you can merge !

**Comment by raphj** [comment]
Yes, I was waiting for the CI to validate my build. It failed, but on an unstable test so all good.

**Pull request merged**